rules/coding-style/src/Rector/String_/UseClassKeywordForClassNameResolutionRector.php

```diff
@@ -50,23 +50,40 @@ public function getNodeTypes(): array
      */
     public function refactor(Node $node): ?Node
     {
-        if (substr_count($node->value, '::') !== 1) {
-            return null;
-        }
-        // a possible static call reference
         // @see https://regex101.com/r/Vv41Qr/1/
-        [$before, $possibleClass, $after] = Strings::split($node->value, '#([\\\\a-zA-Z0-9_\\x80-\\xff]*)::#');
+        $matches = Strings::matchAll($node->value, '#([\\\\a-zA-Z0-9_\\x80-\\xff]*)::#', PREG_PATTERN_ORDER);
+
+        $classNames = array_filter($matches[1], function (string $className) {
+            return class_exists($className);
+        });
 
-        if (! $possibleClass || ! class_exists($possibleClass)) {
-            return null;
+        if (empty($classNames)) {
+            return $node;
         }
 
-        $classConstFetch = new ClassConstFetch(new FullyQualified(ltrim($possibleClass, '\\')), 'class');
+        $classNames = array_map(function (string $className) {
+            return preg_quote($className);
+        }, $classNames);
+
+        // @see https://regex101.com/r/8nGS0F/1
+        $splits = Strings::split($node->value, '#(' . implode('|', $classNames) . ')#');
+
+        foreach ($splits as $split) {
+            if (empty($split)) {
+                continue;
+            }
```

Comment:
Again, empty is dangerous here. Could you think of way how to avoid it completely here?

Reply:
The only way to avoid it completely here is to filter it out with `array_filter` in one of the methods you suggested. Unfortunately String::split() produces empty match group in case the class name is present at the very beginning (and I believe the very end) of the string.

Let me know if it makes sense to you to filter empty match groups out in the newly created method. To me it makes sense.

Reply:
> Let me know if it makes sense to you to filter empty match groups out in the newly created method. To me it makes sense.

Yes :+1: